qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,828 @@
+                    The fitting function should be listed in __fit_funcs__.
+                filter_kwargs: Circuit metadata key and value associated with this curve.
+                    The data points of the curve is extracted from ExperimentData based on
+                    this information.
+                data_option_keys: Circuit metadata keys that are passed to the data processor.
+                    The key should conform to the data processor API.
+
+            See the Examples below for more details.
+        __fit_funcs__: List of callables to fit parameters. This is order sensitive.
+            The list index corresponds to the function index specified by __series__ definition.
+        __param_names__: Name of parameters to fit. This is order sensitive.
+        __base_fitter__: A callable to perform single curve fitting.
+            The function API should conform to the scipy curve fit module.
+
+    Examples:
+
+        T1 experiment
+        =============
+
+        In this type of experiment, the analysis deals with a single curve.
+        Thus filter_kwargs is not necessary defined.
+
+        .. code-block::
+
+            class AnalysisExample(CurveAnalysis):
+
+                __x_key__ = "delay"
+
+                __series__ = [
+                    SeriesDef(
+                        name="t1_decay",
+                        param_names=["a", "tau", "b"],
+                        fit_func_index=0,
+                        filter_kwargs=None,
+                        data_option_keys=["outcome"]
+                    )
+                ]
+
+                __fit_funcs__ = [library.exponential]
+
+                __param_names__ = ["a", "tau", "b"]
+
+        IRB experiment
+        ==============
+
+        In this type of experiment, the analysis deals with two curves.
+        We need a __series__ definition for each curve.
+        Both curves can be represented by the same exponential function,
+        but with a different parameter set. Note that parameters will be partly shared.
+
+        .. code-block::
+
+            class AnalysisExample(CurveAnalysis):
+
+                __x_key__ = "ncliffs"
+
+                __series__ = [
+                    SeriesDef(
+                        name="standard_rb",
+                        param_names=["a", "alpha_std", "b"],
+                        fit_func_index=0,
+                        filter_kwargs={"interleaved": False},
+                        data_option_keys=["outcome"]
+                    ),
+                    SeriesDef(
+                        name="interleaved_rb",
+                        param_names=["a", "alpha_int", "b"],
+                        fit_func_index=0,
+                        filter_kwargs={"interleaved": True},
+                        data_option_keys=["outcome"]
+                    )
+                ]
+
+                __fit_funcs__ = [library.exponential]
```

Comment:
The updated Interleaved RB will need a custom function for the second one to do `f(x, a, alpha_std, alpha_int, b) -> exponential(a, alpha_std * alpha_int, b)`

Reply:
Yes. This setting actually do what you mentioned. The point is to avoid defining `exponential(a, alpha_std * alpha_int, b)` in each subclass, i.e. if we have 10 different curves, we need to define 10 functions - this is not scalable. The base class does the parameter remapping for you. In the IRB example,

series1 defines parameter signature `a, alpha_std, b`
series2 defines parameter signature `a, alpha_int, b`

the fit class defines full signature `a, alpha_std, alpha_int, b`

Thus if we provide a parameter `p0 = 0, 1, 2, 3 = a, alpha_std, alpha_int, b`, the series1 calls `library.exponential(x, 0, 1, 3)` while series2 calls `library.exponential(x, 0, 2, 3)`. This list position can be determined by comparing names of series parameter and entire fit parameters.